### PR TITLE
Fixes `semanticdbVersion` setting so it can be set as `ThisBuild / semanticdbVersion`

### LINKED ISSUE #5110: semanticdbVersion ignored in sbt 1.3.2
## steps
sbt version: 1.3.2

contents of build.sbt
```scala
scalaVersion := "2.13.1"
ThisBuild / semanticdbEnabled := true
ThisBuild / semanticdbVersion := "4.2.3"
ThisBuild / semanticdbIncludeInJar := false
```
## problem
`ThisBuild / semanticdbVersion := "4.2.3"` appears to be ignored and `sbt compile` fails with

```
[error] (update) sbt.librarymanagement.ResolveException: Error downloading org.scalameta:semanticdb-scalac_2.13.1:4.1.0
[error]   Not found
[error]   Not found
[error]   not found: ~/.ivy2/local/org.scalameta/semanticdb-scalac_2.13.1/4.1.0/ivys/ivy.xml
[error]   not found: https://repo1.maven.org/maven2/org/scalameta/semanticdb-scalac_2.13.1/4.1.0/semanticdb-scalac_2.13.1-4.1.0.pom
[error] Total time: 0 s, completed Sep 21, 2019 1:43:43 PM
```

Ie. org.scalameta:semanticdb-scalac_2.13.1:**4.1.0** looks like sbt is attempting to download semanticdb-scalac 4.1.0 which isn't built for Scala 2.13.1, ignoring the semanticdbVersion of 4.2.3 which is built for Scala 2.13.1.

## expectation
`semanticdbVersion` should be respected and the build should compile using semanticdb `4.2.3`.

### PR DESCRIPTION
**Pull request by regadas**
Fixes #5110

### CODE REVIEW

File: main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala [hunk 1]
```diff
@@ -21,14 +21,14 @@ object SemanticdbPlugin extends AutoPlugin {
     semanticdbEnabled := false,
     semanticdbIncludeInJar := false,
     semanticdbOptions := List("-Yrangepos"),
-    semanticdbVersion := "4.1.0",
-    semanticdbCompilerPlugin := {
-      val v = semanticdbVersion.value
-      ("org.scalameta" % "semanticdb-scalac" % v).cross(CrossVersion.full)
-    }
+    semanticdbVersion := "4.1.0"
```
Comment (regadas) [inline main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala]: wondering if we could bump this

Comment (eed3si9n) [reply main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala]: Yes please - https://github.com/sbt/sbt/issues/4633

Comment (regadas) [reply main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala]: Done 👍

File: main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala [hunk 2]
```diff
@@ -17,18 +17,18 @@ object SemanticdbPlugin extends AutoPlugin {
   override def requires = JvmPlugin
   override def trigger = allRequirements
 
-  override lazy val globalSettings = Seq(
+  override lazy val globalSettings: Seq[Def.Setting[_]] = Seq(
     semanticdbEnabled := false,
     semanticdbIncludeInJar := false,
     semanticdbOptions := List("-Yrangepos"),
-    semanticdbVersion := "4.1.0",
+    semanticdbVersion := "4.2.4"
```
Comment (regadas) [inline main/src/main/scala/sbt/plugins/SemanticdbPlugin.scala]: tag exists but wasn't published ... bumping only to `4.2.3`

**Comment by regadas** [comment]
one of the jobs failed weirdly ... @eed3si9n can you re-trigger it?

**Comment by eed3si9n** [comment]
done

**Review by eed3si9n — APPROVED** [review]
Thanks @regadas!

**Pull request merged**